File: app/src/main/java/com/example/lucas/haushaltsmanager/Database/Repositories/Accounts/AccountDAO.java
package com.example.lucas.haushaltsmanager.Database.Repositories.Accounts;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.lucas.haushaltsmanager.Entities.Account;

import java.util.List;
import java.util.UUID;

@Dao
public interface AccountDAO {
    @Query("SELECT * FROM accounts WHERE id = :id")
    Account get(UUID id);

    @Query("SELECT * FROM accounts")
    List<Account> getAll();

    @Insert
    void insert(Account entity);

    @Delete
    void delete(Account account);

    @Update
    void update(Account account);
}

File: app/src/main/java/com/example/lucas/haushaltsmanager/Database/Repositories/AccountDAO.kt
package com.example.lucas.haushaltsmanager.Database.Repositories

import androidx.room.*
import com.example.lucas.haushaltsmanager.entities.Account
import java.util.*

@Dao
interface AccountDAO {
    @Query("SELECT accounts.id, accounts.name, COALESCE(SUM(bookings.price), 0) AS balance FROM accounts LEFT JOIN bookings ON accounts.id = bookings.account_id WHERE accounts.id = :id")
    fun get(id: UUID): Account

    @Query("SELECT accounts.id, accounts.name, COALESCE(SUM(bookings.price), 0) AS balance FROM accounts LEFT JOIN bookings ON accounts.id = bookings.account_id GROUP BY accounts.id")
    fun getAll(): List<Account>

    @Insert
    fun insert(account: Account)

    @Query("SELECT accounts.id, accounts.name, COALESCE(SUM(bookings.price), 0) AS balance FROM accounts LEFT JOIN bookings ON accounts.id = bookings.account_id WHERE accounts.name = :accountName")
    fun getByName(accountName: String): Account

    @Delete
    fun delete(account: Account)

    @Update
    fun update(account: Account)
}